docs/function-wrapping.md

```diff
@@ -0,0 +1,284 @@
+# Function Wrapping
+
+New Relic's agents work largely (but not exclusively) via [code instrumentation](https://en.wikipedia.org/wiki/Instrumentation_(computer_programming)). Agents replace function and method definitions with their own (i.e. monkey patch, wrap, etc.). These new functions and methods will call the original functions and methods they're replacing, but also take steps to time that original function or method call and create New Relic data products (metrics, events, traces, segments, etc) as needed and specified.
+
+The Node.js agent has a leg up on other New Relic agents, in that replacing method definitions is a built in feature of the language.
+
+This document will examine the agent's [Shim.wrap](https://github.com/newrelic/node-newrelic/blob/979994fd7250cad4f19ca4fe65a9a1b2d7ca9311/lib/shim/shim.js#L577-L635) method -- this is the method that other wrappers rely on for swapping out a javascript method definition with their own. Once you understand this method, you'll have the core building block you need to understand all the Shim helper methods related to function wrapping.
+
+## Our Test Program
+
+Normally, in order to use a shim helper you need to configure and setup an [instrumentation](./module-instrumentations.md). You can do that if you like, but this document will use the following test program.
```

Comment:
```suggestion
Normally, in order to use a shim helper you need to configure and setup an [instrumentation](./docs/module-instrumentations.md). You can do that if you like, but this document will use the following test program.
```

Reply:
Curious why you are suggesting this change? The link works for me testing directly from branch and open the file via this PR. Will it behave different once merged?

Reply:
Ah, I see what's up, I tried link in preview mode on PR review. Link looks good on second look because referencing file is in same folder. Removing suggestion. 😸